examples/reduction/examples/matmul.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2023 NVIDIA Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import argparse
+
+import cunumeric as np
+from reduction import (
+    matmul,
+    multiply,
+    print_store,
+    sum_over_axis,
+    to_cunumeric_array,
```

Comment:
Personal preference would be to only have a `print_store` function and that be the only place we need a `to_cunumeric_array` function. It seems more intuitive to me to create cunumeric arrays and fetch stores from them rather than create a store and then convert it to a cunumeric array.

Reply:
In general, it would probably be better to to implement the `__str__` function of the store (wrapper) so that `print(store)` would yield the desired result.